### ISSUE TITLE
Breaking change SessionExtensions.GetTemporarySessionId

### ISSUE THREAD

**Issue by mjdaly** [ServiceStack/Issues#399]
Upon switching from SS 4.0.40 to 4.0.52, `SessionExtensions.GetTemporarySessionId` has changed behaviour, it is a breaking change.

Using a `JsonServiceClient` with `AlwaysSendBasicAuthHeader = true`, first requests have access to the authenticated session, subsequent requests get blank sessions.

It looks like `GetSessionParam` is not returning the correct session id from cookies in the request.

In the example, you can see differing results from two requests made.
[TestSSAuth.zip](https://github.com/ServiceStack/Issues/files/112449/TestSSAuth.zip)

**Comment by mythz** [comment]
The issue is due to the `GenerateNewSessionCookiesOnAuthentication` change [documented in the v4.0.44 release notes](https://github.com/ServiceStack/ServiceStack/blob/master/docs/2015/release-notes.md#generate-new-session-cookies-on-authentication).

That you can disable with:

```csharp
var auth = new AuthFeature(() => new CustomUserSession(),
    new IAuthProvider[]
    {
        new CustomBasicAuthProvider(),
        new CredentialsAuthProvider(appSettings) {SessionExpiry = TimeSpan.FromMinutes(30)}
    },
    "/login")
{
    GenerateNewSessionCookiesOnAuthentication = false,
};
```

Still thinking about how this should work with Basic Auth.

**Comment by mythz** [comment]
ok the default configuration where `GenerateNewSessionCookiesOnAuthentication=true` should now also work with this commit: https://github.com/ServiceStack/ServiceStack/commit/3e18cf6de162991060008fa4a5271ce715bd3ba6

Basically the new session in the re-auth request wasn't getting saved after generating new session keys.

This change is now available from v4.0.53 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

**Comment by G-Shriya** [comment]
Facing the same issue even while using ServiceStack version 4.5
